core/storage/email/gae_models.py

```diff
@@ -185,7 +185,7 @@ def get_by_hash(cls, email_hash, sent_datetime_lower_bound=None):
         return messages
 
     @classmethod
-    def _generate_hash(cls, recipient_id, email_subject, email_body):
+    def generate_hash(cls, recipient_id, email_subject, email_body):
```

Comment:
In general you should try not to just change private methods to public just because those methods are currently tested. This changes the API that's exposed by a class, which is a big change. It is better to try and find a way to test these private methods through the public API.

In this case, it looks like this hash is generated in put(). So perhaps you can write equivalent tests that create and save SentEmailModel instances, then retrieve their hashes by getting them from the datastore again?

Reply:
Ok, thanks, I will try do that!

Reply:
Could I also create a class called GenerateHash and make this function public in this class, or isn't that a good idea?

Reply:
Could you explain which other classes use this method? (Test classes do not count.)

Reply:
You're right, no non-test class uses this method, so creating another class doesn't really make sense. Thanks for the explanations, I will try to do what you suggested. :)